src/libraries/System.Private.CoreLib/src/System/Threading/LowLevelLifoSemaphore.cs

```diff
@@ -39,6 +39,9 @@ public LowLevelLifoSemaphore(int initialSignalCount, int maximumSignalCount, int
 
         public bool Wait(int timeoutMs, bool spinWait)
         {
+#if TARGET_BROWSER && FEATURE_WASM_THREADS
+            ThrowIfInvalidSemaphoreKind(LifoSemaphoreKind.Normal);
+#endif
```

Comment:
Are there places where we use `LowLevelLifoSemaphore` that we're keeping on the old "call `Wait`" plan? If not, can we instead `#ifdef`-out the implementation of `Wait` or make it always throw `PlatformNotSupportedException` in this configuration instead of introducing two separate modes that make different sets of method calls valid?

Reply:
We need both, I think. There are legitimate uses for the synchronous version on the browser.l, too. Just not in the threadpool worker interop scenario.

I could instead do something more OO-y with a common base class that handles the Counters logic. Wasn't sure if it was worthwhile.

Reply:
I'd prefer that if there isn't a performance issue from doing so. But I know that this is a high perf area, so if it's not possible that's okay.

Reply:
Made a common base class `LowLevelLifoSemaphoreBase` and two subclasses `LowLevelLifoSemaphore` (same as before) and `LowLevelLifoAsyncWaitSemaphore` (with `PrepareAsyncWait` instead of `Wait`).